packages/sdk/tests/test-voucher-utils.ts

```diff
@@ -0,0 +1,388 @@
+    ),
+    faucetWallet: new Wallet(
+      '0xde43b282c2931fc41ca9e1486fedc2c45227a3b9b4115c89d37f6333c8816d89'
+    ),
+    voucherHubAddress: VOUCHER_HUB_ADDRESS, // TODO: change with deployment address once voucher is deployed on bellecour
+    voucherManagerWallet: new Wallet(
+      '0x2c906d4022cace2b3ee6c8b596564c26c4dcadddf1e949b769bcb0ad75c40c33'
+    ),
+    voucherSubgraphURL: DRONE
+      ? 'http://gaphnode:8000/subgraphs/name/bellecour/iexec-voucher'
+      : 'http://localhost:8000/subgraphs/name/bellecour/iexec-voucher',
+    debugWorkerpool: 'debug-v8-bellecour.main.pools.iexec.eth',
+    debugWorkerpoolOwnerWallet: new Wallet(
+      '0x800e01919eadf36f110f733decb1cc0f82e7941a748e89d7a3f76157f6654bb3'
+    ),
+    prodWorkerpool: 'prod-v8-bellecour.main.pools.iexec.eth',
+    prodWorkerpoolOwnerWallet: new Wallet(
+      '0x6a12f56d7686e85ab0f46eb3c19cb0c75bfabf8fb04e595654fc93ad652fa7bc'
+    ),
+    provider: new JsonRpcProvider(
+      DRONE ? 'http://bellecour-fork:8545' : 'http://localhost:8545'
+    ),
+    defaults: {
+      hubAddress: '0x3eca1B216A7DF1C7689aEb259fFB83ADFB894E7f',
+      ensRegistryAddress: '0x5f5B93fca68c9C79318d1F3868A354EE67D8c006',
+      ensPublicResolverAddress: '0x1347d8a1840A810B990d0B774A6b7Bb8A1bd62BB',
+      isNative: true,
+      useGas: false,
+      name: 'bellecour',
+    },
+    isAnvil: true,
+  },
+};
+
+const faucetSendWeiToReachTargetBalance =
+  (chain) =>
+  async (address, targetWeiBalance, tryCount = 1) => {
+    const currentBalance = await chain.provider.getBalance(address);
+    const delta = BigInt(`${targetWeiBalance}`) - currentBalance;
+    if (delta < 0n) {
+      console.warn(
+        `Faucet send Eth: aborted - current balance exceed target balance`
+      );
+      return;
+    }
+    try {
+      const tx = await chain.faucetWallet
+        .connect(chain.provider)
+        .sendTransaction({ to: address, value: delta });
+      await tx.wait();
+    } catch (e) {
+      console.warn(`Faucet send Eth: error (try count ${tryCount}) - ${e}`);
+      // retry as concurrent calls can lead to nonce collisions on the faucet wallet
+      if (tryCount < 3) {
+        await sleep(3000 * tryCount);
+        await faucetSendWeiToReachTargetBalance(chain)(
+          address,
+          targetWeiBalance,
+          tryCount + 1
+        );
+      } else {
+        throw Error(`Failed to send Eth from faucet (tried ${tryCount} times)`);
+      }
+    }
+  };
+
+export const setBalance = (chain) => async (address, targetWeiBalance) => {
+  if (chain.isAnvil) {
+    await fetch(chain.rpcURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        method: 'anvil_setBalance',
+        params: [address, ethers.toBeHex(targetWeiBalance)],
+        id: 1,
+        jsonrpc: '2.0',
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  } else {
+    await faucetSendWeiToReachTargetBalance(chain)(address, targetWeiBalance);
```

Comment:
will never be used, the whole `faucetSendWeiToReachTargetBalance` implementation can be removed